config/routes.rb

```diff
@@ -3,8 +3,17 @@
   # root "articles#index"
   # Defines the root path route ("/")
 
-  root "users#index"
-  resources :users, only: [:show]
+  root "users#index", as: "users_home"
+
+  resources :users, only: [:show] do
+    resources :posts, only: [:new, :create]
+    resources :likes, only: [:create]
+    resources :comments, only: [:create]
+  end
+
   get "users/:user_id/posts", to: "posts#index"
-  get "users/:user_id/posts/:post_id", to: "posts#show"
+  get "users/:user_id/posts/:post_id", to: "posts#show", as: "posts_show"
+
+  get "users/:user_id/posts/:post_id/comment/new", to: "comments#new" #new
+
```

Comment:
[OPTIONAL]
- [ ] As suggested by the previous reviewer, This `route.rb` can be refactored further like so:
```rb
resources :users, only: [:index, :show] do
  resources :posts, only: [:new, :create, :show]
 # ...
end
```